plugins/www/Optimove.js

```diff
@@ -11,6 +11,28 @@ document.addEventListener("deviceready", init, false);
 document.addEventListener("resume", resume, false);
 document.addEventListener("pause", pause, false);
 
+function isString(val) {
+  return typeof val === "string";
+}
+
+function isNonEmptyString(val) {
+  return val && isString(val);
+}
+
+function isFunction(param) {
+  return typeof param === "function";
+}
+
+function isInboxItemValid(inboxItem) {
+  var mandatoryValues = ["id", "title", "subtitle", "sentAt"];
```

Comment:
why specifically these fields? inbox item has more, but the only required is id. + with these fields example app will stop working as in `getInboxItemForTesting` we only give id to a fake inbox item.

Will this become redundant when we convert it to typescript? So, perhaps 
1. remove this validation function
2. convert index.js and Optimove.js to typescript
3. update `getInboxItemForTesting` to match typescript interface